File: csharp/GetCryptoSnapshot.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using Intrinio.SDK.Api;
using Intrinio.SDK.Client;
using Intrinio.SDK.Model;

namespace Example
{
  public class GetCryptoSnapshot
  {
    public static void Main()
    {
      Configuration.Default.AddApiKey("api_key", "YOUR_API_KEY");

      var cryptoApi = new CryptoApi();
      var pair = "btcusd";  // string | Return the snapshot for the given Crypto Currency Pair. (optional) 
      var exchange = "gemini";  // string | Return the snapshot for a Crypto Currency on the given Crypto Exchange. (optional) 
      var currency = "BTC";  // string | Return the snapshot for the given Crypto Currency. (optional) 

      try
      {
        ApiResponseCryptoSnapshot result = cryptoApi.GetCryptoSnapshot(pair, exchange, currency);
        CryptoPairSummary crypto_pair = result.Pair;
        CryptoExchangeSummary crypto_exchange = result.Exchange;
        CryptoSnapshot snapshot = result.Snapshot;

        Console.WriteLine("Snapshot for " + crypto_pair.Name + " on " + crypto_exchange.Name + "!");
        Console.WriteLine();
        Console.WriteLine("Last updated:     " + snapshot.LastUpdated);
        Console.WriteLine("Bid:              " + snapshot.Bid);
        Console.WriteLine("Bid size:         " + snapshot.BidSize);
        Console.WriteLine("Ask:              " + snapshot.Ask);
        Console.WriteLine("Ask size:         " + snapshot.AskSize);
        Console.WriteLine("Change:           " + snapshot.Change);
        Console.WriteLine("Change percent:   " + snapshot.ChangePercent);
        Console.WriteLine("Volume:           " + snapshot.Volume);
        Console.WriteLine("Open:             " + snapshot.Open);
        Console.WriteLine("High:             " + snapshot.High);
        Console.WriteLine("Low:              " + snapshot.Low);
        Console.WriteLine("Last trade side:  " + snapshot.LastTradeSide);
        Console.WriteLine("Last trade time:  " + snapshot.LastTradeTime);
        Console.WriteLine("Last trade price: " + snapshot.LastTradePrice);
        Console.WriteLine("Last trade size:  " + snapshot.LastTradeSize);
      }
      catch (Exception e)
      {
        Debug.Print("Exception when calling CryptoApi.GetCryptoSnapshot: " + e.Message);
      }
    }
  }
}

File: python/get_crypto_snapshot.py
from __future__ import print_function
import time
import intrinio_sdk
from intrinio_sdk.rest import ApiException
from pprint import pprint
 
intrinio_sdk.ApiClient().configuration.api_key['api_key'] = 'YOUR_API_KEY'
 
crypto_api = intrinio_sdk.CryptoApi()

pair = 'btcusd' # str | Return the snapshot for the given Crypto Currency Pair. (optional)
exchange = 'gemini' # str | Return the snapshot for a Crypto Currency on the given Crypto Exchange. (optional)
currency = 'BTC' # str | Return the snapshot for the given Crypto Currency. (optional)
 
try:
    api_response = crypto_api.get_crypto_snapshot(pair=pair, exchange=exchange, currency=currency)
    snapshot = api_response.snapshot
    pair_name = api_response.pair.name
    exchange_name = api_response.exchange.name

    print("Snapshot for %s on %s!" % (pair_name, exchange_name))
    print("------------------------------------------------------")
    print("Last updated:     %s" % (snapshot.last_updated))
    print("Bid:              %s" % (snapshot.bid))
    print("Bid size:         %s" % (snapshot.bid_size))
    print("Ask:              %s" % (snapshot.ask))
    print("Ask size:         %s" % (snapshot.ask_size))
    print("Change:           %s" % (snapshot.change))
    print("Change percent:   %s" % (snapshot.change_percent))
    print("Volume:           %s" % (snapshot.volume))
    print("Open:             %s" % (snapshot.open))
    print("High:             %s" % (snapshot.high))
    print("Low:              %s" % (snapshot.low))
    print("Last trade side:  %s" % (snapshot.last_trade_side))
    print("Last trade time:  %s" % (snapshot.last_trade_time))
    print("Last trade price: %s" % (snapshot.last_trade_price))
    print("Last trade size:  %s" % (snapshot.last_trade_size))
except ApiException as e:
    print("Exception when calling CryptoApi->get_crypto_snapshot: %s\n" % e)
